include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+#endif
+        else
+            _program.clear();
+    }
+#else // windows
+    void _getProgram()
+    {
+        _program = "powershell.exe -NoLogo -NonInteractive -Command \"& {Invoke-WebRequest -erroraction 'silentlycontinue' -OutFile";
+    }
+#endif
+
+    void _updateCommand()
+    {
+        if (!_program.empty())
+        {
+            _command = _program + " " + _path + "/" + _name + "_version.txt " + _url + _os + "_64_"+ _name + "_" + _version;
+#if defined(PLATFORM_WINDOWS)
+            _command = _command + "; exit  [int] -not $?}\" > nul 2>&1";
+#endif
+        }
+    }
+};
+
+// ==========================================================================
+// Metafunctions
+// ==========================================================================
+
+// ==========================================================================
+// Functions
+// ==========================================================================
+
+// ----------------------------------------------------------------------------
+// Function setURL()
+// ----------------------------------------------------------------------------
+inline void setURL(VersionCheck & me, std::string url)
+{
+    std::swap(me._url, url);
+    me._updateCommand();
+}
+
+// ----------------------------------------------------------------------------
+// Function _checkWritability()
+// ----------------------------------------------------------------------------
+#if defined(PLATFORM_WINDOWS)
+inline bool _checkWritability(std::string path)
+{
+    DWORD ftyp = GetFileAttributesA(path.c_str());
+    if (ftyp == INVALID_FILE_ATTRIBUTES || !(ftyp & FILE_ATTRIBUTE_DIRECTORY))
+    {
+        if (!CreateDirectory(path.c_str(), NULL))
+            return false;
+    }
+
+    HANDLE dummyFile; // check writablity by trying to create a file in GENERIC_WRITE mode
+    std::string fileName = path + "/dummy.txt";
+    dummyFile = CreateFile(fileName.c_str(), GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+
+    if (dummyFile == INVALID_HANDLE_VALUE)
+        return false;
+    
+    CloseHandle(dummyFile);
+    DeleteFile(fileName.c_str());
+    return true;
+}
+#else
+inline bool _checkWritability(std::string path)
+{
+    struct stat d_stat;
+    if (stat(path.c_str(), &d_stat) < 0)
+    {
+        // try to make dir
+        std::string makeDir = "mkdir -p " + path;
+        if (system(makeDir.c_str()))
+            return false; // could not create home dir
+
+        if (stat(path.c_str(), &d_stat) < 0) // repeat stat
+            return false;
+    }
+
+    if (!(d_stat.st_mode & S_IWUSR))
+        return false; // dir not writable
+
+    return true;
+}
+#endif
+
+// ----------------------------------------------------------------------------
+// Function _getFileTimeDiff()
+// ----------------------------------------------------------------------------
+#if defined(PLATFORM_WINDOWS)
+inline double _getFileTimeDiff(std::string version_file)
```

Comment:
Is it not easier to write the timestamp into the file, without all this kernel voodoo?